libraries/SPI/SPI.cpp

```diff
@@ -531,48 +528,90 @@ void SPIClass::transferBytes(const uint8_t * out, uint8_t * in, uint32_t size) {
 }
 
 /**
- * Note:
- *  in and out need to be aligned to 32Bit
- *  or you get an Fatal exception (9)
  * @param out uint8_t *
  * @param in  uint8_t *
  * @param size uint8_t (max 64)
  */
 void SPIClass::transferBytes_(const uint8_t * out, uint8_t * in, uint8_t size) {
-    while(SPI1CMD & SPIBUSY) {}
-    // Set in/out Bits to transfer
+    while(SPI1CMD & SPIBUSY) {} // Make sure we're IDLE on the SPI
 
-    setDataBits(size * 8);
+    int bytesToTransfer = size; // How much to read/write from the FIFO in bytes
 
-    volatile uint32_t * fifoPtr = &SPI1W0;
-    uint8_t dataSize = ((size + 3) / 4);
+    if (out && !in) {
+        int bytesLeft = size;
 
-    if(out) {
-        uint32_t * dataPtr = (uint32_t*) out;
-        while(dataSize--) {
-            *fifoPtr = *dataPtr;
-            dataPtr++;
-            fifoPtr++;
+        // Only transmitting, get out 32b aligned
+        while (bytesLeft && ((uint32_t)out & 3)) {
+            SPI.transfer(*(out++));
```

Comment:
What? Did you mean just:
`transfer(*(out++));`
?

Reply:
Refactored to remove this bit completely.